dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -86,6 +89,35 @@ private static DependencyNode resolveCompileTimeDependencies(Artifact artifact)
     return node;
   }
 
+  private static DependencyNode resolveCompileTimeDependencies(List<Artifact> artifacts)
+      throws DependencyCollectionException, DependencyResolutionException {
+    String key = artifacts.stream().map(Artifacts::toCoordinates).collect(Collectors.joining(","));
+    if (cache.containsKey(key)) {
+      return cache.get(key);
```

Comment:
Will we ever encounter a key more than once that is created by joining artifacts by comma?

Reply:
But if the cache is by individual artifacts, we could catch all the artifacts in the cache anyway, right? If we put comma-joined keys in, then we'd have the same artifacts represented in a bunch of different ways - the cache would be bigger and more complex to reason about.

Reply:
Let me remove this cache. I don't see much use case either. We can implement one when necessary.